### PR TITLE
VGNWB-145 (Elastic search)

### PR DESCRIPTION
**Pull request by erhkamal**
Hi Erick,

All changes corresponding to elastic search have been merged to VGNWB-145.
This needs to be merged with develop.

Thanks,
Kamal

### CODE REVIEW

File: annotations/views.py [hunk 1]
```diff
@@ -912,3 +915,24 @@ def concept_details(request, conceptid):
     })
 
     return HttpResponse(template.render(context))
+
+
+class TextSearchView(SearchView):
```
Comment (erickpeirson) [inline annotations/views.py:1291]: Don't forget to add a docstring

File: annotations/views.py [hunk 2]
```diff
@@ -912,3 +915,24 @@ def concept_details(request, conceptid):
...
+    template = 'templates/search/search.html'
+    queryset = SearchQuerySet()
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(TextSearchView, self).get_context_data(*args, **kwargs)
+        return context
+
+    def form_valid(self, form):
```
Comment (erickpeirson) [inline annotations/views.py:1301]: Don't forget to add a docstring

File: annotations/views.py [hunk 3]
```diff
@@ -912,3 +915,24 @@ def concept_details(request, conceptid):
     })
 
     return HttpResponse(template.render(context))
+
+
+class TextSearchView(SearchView):
+    template = 'templates/search/search.html'
+    queryset = SearchQuerySet()
+
+    def get_context_data(self, *args, **kwargs):
```
Comment (erickpeirson) [inline annotations/views.py:1296]: Don't forget to add a docstring

File: annotations/search_indexes.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import datetime
+from haystack import indexes
+from annotations.models import Text
+
+class TextIndex(indexes.SearchIndex, indexes.Indexable):
```
Comment (erickpeirson) [inline annotations/search_indexes.py:5]: Don't forget to add a docstring

File: annotations/search_indexes.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    text = indexes.CharField(document=True, use_template=True)
+    title = indexes.CharField(model_attr='title', indexed=False)
+    addedBy = indexes.CharField(model_attr='addedBy')
+    uri = indexes.CharField(model_attr='id')
+    relation_count = indexes.CharField(model_attr='relation_count')
+
+    def get_model(self):
```
Comment (erickpeirson) [inline annotations/search_indexes.py:13]: Don't forget to add a docstring

File: annotations/search_indexes.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        return Text
+
+    def index_queryset(self, using=None):
```
Comment (erickpeirson) [inline annotations/search_indexes.py:17]: Don't forget to add a docstring

File: vogon/urls.py [hunk 7]
```diff
@@ -21,6 +21,7 @@
 from rest_framework import routers
 from rest_framework_nested import routers as nrouters
 from annotations import views
+from haystack.generic_views import SearchView
```
Comment (erickpeirson) [inline vogon/urls.py]: Are you using `SearchView` anywhere in `urls.py`?

**Comment by erhkamal** [comment]
Hi @erickpeirson,

We have incorporated your review comments.
Could you please review once more and merge it to develop?

Thanks,
Kamal

**Pull request merged**